include/arch/arc/syscall.h

```diff
@@ -182,7 +183,7 @@ static inline int _arch_is_user_context(void)
 			 : "=r"(status)
 			 : [status32] "i" (_ARC_V2_STATUS32));
 
-	return !(status & _ARC_V2_STATUS32_US);
+	return (status & _ARC_V2_STATUS32_US) ? true : false;
```

Comment:
I believe you missed the ! on the original line, please swap true and false (and, would be good to check effiency of generated code to ensure no performance penalty).

Reply:
yeah, thanks for pointing it out.